File: katas/7 kyu/square-every-digit/solution.py
import codewars_test as test
# -------------------------------- MY SOLUTION ------------------------------- #
def square_digits(num) : return int(''.join(str(int(i)**2) for i in str(num)))
# ------------------------------ CLEVER SOLUTION ----------------------------- #
square_digits = lambda num: int(''.join(str(int(d)**2) for d in str(num)))
# ----------------------------------- TEST ----------------------------------- #
@test.describe("Premade tests: ")
def fixed_tests():
    @test.it('Basic Test Cases')
    def basic_test_cases():
      test.assert_equals(square_digits(9119), 811181)
      test.assert_equals(square_digits(0), 0)

File: katas/7 kyu/square-every-digit/solution.php
<!-- square-every-digit -->
<?php
use PHPUnit\Framework\TestCase;
/* ------------------------------- MY SOLUTION ------------------------------ */
function square_digits($num){
  $num = str_split(strval($num) , 1);
  $r = '';
  foreach ($num as $digit) {
    $r .= strval(intval($digit**2));
  }
  return intval($r);
}
/* ---------------------------------- TEST ---------------------------------- */
class MyTestCases extends TestCase
{
    // test function names should start with "test"
    public function testBasics() {
      $this->assertEquals(square_digits(9119), 811181);
      $this->assertEquals(41636640, square_digits(24680));
      $this->assertEquals(19254981, square_digits(13579));
      $this->assertEquals(square_digits(0), 0);
    }
}